operators/src/adapters/raster_overlap_adapter.rs

```diff
@@ -282,21 +325,200 @@ pub fn insert_projected_pixels<'a, T: Pixel, I: Iterator<Item = &'a (GridIdx2D,
     Ok(())
 }
 
+/// This trait defines the behavior of the `RasterOverlapAdapter`.
+pub trait TileSubQuery<T>: Send
+where
+    T: Pixel,
+{
+    type FoldFuture: TryFuture<Ok = (RasterTile2D<T>, Self::FoldCompanion), Error = error::Error>;
+    type FoldMethod: Clone
+        + Fn((RasterTile2D<T>, Self::FoldCompanion), RasterTile2D<T>) -> Self::FoldFuture;
+    type FoldCompanion: Clone + Send;
+
+    /// The no-data-value to use in the resulting `RasterTile2D`
+    fn result_no_data_value(&self) -> Option<T>;
+    /// The initial fill-value of the accumulator (`RasterTile2D`).
+    fn initial_fill_value(&self) -> T;
+
+    /// This method geneerates a new accumulator which is used to fold the `Stream` of `RasterTile2D` of a sub-query.
+    fn new_fold_accu(
+        &self,
+        tile_info: TileInformation,
+        query_rect: QueryRectangle,
+    ) -> Result<(RasterTile2D<T>, Self::FoldCompanion)>;
```

Comment:
Maybe we can even introduce a struct for this that contains the two types. This way, you don't have to de-structure it each time. Since it is a generic aggregator and no a specific operator, I would think about doing this.

Reply:
i don't see why this is better then using a tuple. i can also do tuple.0 and tuple.1  instead of de-structuring if you want...

Reply:
I think it is nicer to used instead of requiring to de-structure or to say .0 or .1.

Reply:
but it is just a tuple. thats why the tuple type exists...